feat: add docs for Jenkins metrics and traces to Signoz (#1393)
SigNoz/signoz.io: data/docs/cicd/jenkins/jenkins-tracing.mdx

---
date: 2025-04-28
id: jenkins-tracing
title: Jenkins Traces
description: Instrument Jenkins with OpenTelemetry and send traces to SigNoz
---
   
## Overview 
This integration helps you to monitor and observe Jenkins with OpenTelemetry by visualizing jenkins jobs and pipelines executions as distributed traces in SigNoz
To know more about tracing refer [here](https://signoz.io/docs/instrumentation/overview/)

## Prerequisites

> - Jenkins [>2.479.3]

Reviewer: ```suggestion
- Jenkins (>2.479.3)
- Jenkins pipelines that you want to monitor
- Administrative access to Jenkins to install plugins
```

Reviewer: @Apurv11 please resolve

> ### Setup

Reviewer: ```suggestion
## Setup
```

> SigNoz runs its own OTel collector in both the Self-Hosted and Cloud version

Reviewer: ```suggestion
```

Reviewer: @Apurv11 please resolve

> ### Install and Configure Jenkins OpenTelemetry plugin

Reviewer: ```suggestion
### Step 1: Install and Configure Jenkins OpenTelemetry plugin
```

Reviewer: @Apurv11 please resolve

> * For installation refer [here](https://plugins.jenkins.io/opentelemetry/)

Reviewer: ```suggestion
* For installing the plugin, please refer to [this documentation](https://plugins.jenkins.io/opentelemetry-agent-metrics/).
```

Reviewer: @Apurv11 please resolve

<Tabs>

<TabItem value="signoz-cloud" label="SigNoz Cloud" default>

> ### Step 1: Set the OTLP Endpoint

Reviewer: ```suggestion
### Step 2: Set the OTLP Endpoint
```

Reviewer: @Apurv11 please resolve

Navigate to Manage Jenkins -> System -> OpenTelemetry section

Depending on the choice of your region for the SigNoz cloud, the ingest endpoint will vary according to this table.
  | Region | Endpoint |
  | --- | --- |
  | US |  ingest.us.signoz.cloud:443 |
  | IN |  ingest.in.signoz.cloud:443 |

>   | EU | ingest.eu.signoz.cloud:443 |

Reviewer: ```suggestion
Navigate to **Manage Jenkins** -> **System** -> **OpenTelemetry** section

In the OTLP Endpoint section, add your SigNoz ingestion endpoint `ingest.<region>.signoz.cloud`

Set the <region> to match your SigNoz Cloud [region](https://signoz.io/docs/ingestion/signoz-cloud/overview/#endpoint).

```

Reviewer: Include the screenshot showing the OTLP endpoint section highlighted using a orange colored box. Can figma/canva or any other tool for this.

Reviewer: @Apurv11 please add details about the endpoint and region as shown in the above comment.

### Step 2: Set the Authentication headers
In the same section click on Authentication and select **Header Authentication**

> Add Header key as **signoz-ingestion-key** and value as SigNoz_Ingestion_Key

Reviewer: ```suggestion
Add Header key as **signoz-ingestion-key** and value as `<your-ingestion-key>`.

Replace <your-ingestion-key> with your SigNoz [ingestion key](https://signoz.io/docs/ingestion/signoz-cloud/keys/).

```

Reviewer: @Apurv11 please resolve

<figure data-zoomable align="center">
    <img
        src="/img/docs/cicd/jenkins/jenkins-otel-cloud-config.webp"
        alt="Jenkins OTel plugin configuration for SigNoz cloud"
    />
    <figcaption>
        <i>
        Jenkins OTel plugin configuration for SigNoz cloud
        </i>
    </figcaption>

> </figure>

Reviewer: Please work on changing this image as discussed.

> Note: Replace \<signoz-ingestion-key\> with your ingestion key

Reviewer: ```suggestion
```

Reviewer: @Apurv11 please resolve

</TabItem>

<TabItem value='self-host' label='Self-Host' default>

Navigate to Manage Jenkins -> System -> OpenTelemetry section 
<figure data-zoomable align="center">
    <img
        src="/img/docs/cicd/jenkins/jenkins-otel-config.webp"
        alt="Jenkins OTel plugin configuration for Self-Hosted SigNoz"
    />
    <figcaption>
        <i>
        Jenkins OTel plugin configuration for Self-Hosted SigNoz
        </i>
    </figcaption>
</figure>

> Enter the SigNoz self-hosted endpoint in the textbox and set **No Authentication**

Reviewer: ```suggestion
Enter the SigNoz self-hosted endpoint in the OTLP endpoint textbox and set **No Authentication**
```

Reviewer: @Apurv11 please resolve

</TabItem>
</Tabs>

> ## Visualize the traces in SigNoz Traces section

Reviewer: ```suggestion
## Visualize Traces in SigNoz
```
In your SigNoz UI, head over to the Traces Tab, and you should be able to see your traces.
You can use the [trace explorer](https://signoz.io/docs/product-features/trace-explorer/) to filter and analyze your traces.

<figure data-zoomable align="center">
    <img
        src="/img/docs/cicd/jenkins/jenkins-traces.webp"
        alt="Jenkins pipelines traces on SigNoz UI"
    />
    <figcaption>
        <i>
        Jenkins pipelines traces on SigNoz UI
        </i>
    </figcaption>
</figure>

> ## Jenkinsfile custom spans
> 
> With the OTel plugin installed, you can also configure custom spans in your Jenkins pipelines
> 
> ```groovy
>   pipeline {
>     agent any
>     stages {
>       stage('Add Custom Span Attribute') {
>           steps {
>             script {
>               withSpanAttribute(
>                   key: 'my-error-prone-span',
>                   value: 'new-traces',
>                   type: 'STRING',
>                   target: 'CURRENT_SPAN'
>               )
>               echo 'Custom span attribute added'
>             }
>           }
>         }
>     }
>   }
> ```

Reviewer: ```suggestion
## Custom Spans in Jenkinsfile

With the OTel plugin installed, you can enhance your Jenkins pipelines with custom observability by adding span attributes, creating custom spans, and enriching trace data. 

### Adding Custom Span Attributes

The following example shows how to add custom attributes to the current span within your Jenkins pipeline:

```groovy
    pipeline {
       agent any
       stages {
           stage('Add Custom Span Attribute') {
               steps {
                   script {
                       withSpanAttribute(
                           key: 'my-error-prone-span',
                           value: 'new-traces',
                           type: 'STRING',
                           target: 'CURRENT_SPAN'
                       )
                       echo 'Custom span attribute added'
                   }
               }
           }
       }
    }
```
**What this code does:**

* `withSpanAttribute()`: A Jenkins OTel plugin function that adds custom key-value metadata to spans
* `key: 'my-error-prone-span'`: The attribute name that will appear in your trace data
* `value: 'new-traces'`: The attribute value associated with the key
* `type: 'STRING'`: Specifies the data type of the attribute (can be STRING, LONG, DOUBLE, or BOOLEAN)
* `target: 'CURRENT_SPAN'`: Indicates that the attribute should be added to the currently active span (the stage span in this case)
```

Please verify the above content thoroughly especially the code block explanation part.

> For more details about spans refer [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span)

Reviewer: ```suggestion
```

> ## Conclusion
> 
> Jenkins Otel plugin can be used to visualize Jenkins traces effectively on SigNoz

Reviewer: ```suggestion
```

---
date: 2025-04-28
id: jenkins-tracing

> title: Jenkins Tracing with OpenTelemetry on Signoz via Open Telemetry plugin 

Reviewer: ```suggestion
title: Jenkins Tracing 
```

description: Instrument Jenkins with OpenTelemetry and send traces to SigNoz
---
   
## Overview 
This integration helps you to monitor and observe Jenkins with OpenTelemetry by visualizing jenkins jobs and pipelines executions as distributed traces.

This document will show you how to:
- configure the Jenkins OpenTelemetry plugin
- Configure the Signoz Otel collector
- Visualize the traces in Signoz Traces section 

To know more about tracing refer [here](../../traces-management/guides)

## Prerequisites

- Jenkins [>2.479.3]
- SigNoz Cloud(optional)
  - endpoint URL

>   - api key

Reviewer: ```suggestion

```
